### PR TITLE
Create Grafana Dashboard for Nessie service

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: grafana/nessie.json [hunk 1]
```diff
@@ -0,0 +1,4971 @@
...
+      
+      "gridPos": {
+        "h": 1,
+        "w": 24,
+        "x": 0,
+        "y": 108
+      },
+      "id": 163,
+      "panels": [],
+      "title": "HTTP Edpoints",
```
Comment (XN137) [inline grafana/nessie.json:4472]: typo :)

File: grafana/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+2. Install Prometheus
+    ```sh
+    helm -n nessie-dev install prometheus prometheus-community/kube-prometheus-stack
+    ```
+3. Install Nessie (see also https://github.com/projectnessie/nessie/blob/main/helm/README.md)
+    ```sh
+      helm repo add nessie-helm https://charts.projectnessie.org
+      helm repo update
+      helm install --namespace nessie-dev nessie nessie-helm/nessie --set 'serviceMonitor.labels.release=prometheus'
+      ```
```
Comment (XN137) [inline grafana/README.md:31]: indent is off

File: grafana/nessie.json [hunk 3]
```diff
@@ -0,0 +1,4971 @@
...
+      
+      "gridPos": {
+        "h": 1,
+        "w": 24,
+        "x": 0,
+        "y": 84
+      },
+      "id": 131,
+      "panels": [],
+      "title": "Garbage Collection",
```
Comment (XN137) [inline grafana/nessie.json:3511]: should we add `JVM` prefix here? to not confuse with nessie GC operations

**Review by snazy — COMMENTED** [review]
IIRC there's some "marketplace" for predefined Grafana metrics/dashboards. Should we maybe use that instead of adding a 4000 line json file - and provide some base Grafana configuration using that externally maintained dashboard?

**Comment by nastra** [comment]
> IIRC there's some "marketplace" for predefined Grafana metrics/dashboards. Should we maybe use that instead of adding a 4000 line json file - and provide some base Grafana configuration using that externally maintained dashboard?

This dashboard is already based off of https://grafana.com/grafana/dashboards/14370

File: grafana/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Prometheus & Grafana for Nessie
+
+## Dev installation for local experiments
+
+These steps are mainly for setting up Prometheus/Grafana/Nessie locally.
+
+* Install Minikube as described in https://minikube.sigs.k8s.io/docs/start/
+* Install Helm as described in https://helm.sh/docs/intro/install/
```
Comment (XN137) [inline grafana/README.md:8]: maybe also link https://kubernetes.io/docs/tasks/tools/ for `kubectl` if we act like this is for users starting from 0 (using the kubectl alias from minikube did not work with the cmds below).

File: grafana/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+```
+3. Install Nessie (see also https://github.com/projectnessie/nessie/blob/main/helm/README.md)
+```sh
+helm repo add nessie-helm https://charts.projectnessie.org
+helm repo update
+helm install --namespace nessie-dev nessie nessie-helm/nessie --set 'serviceMonitor.labels.release=prometheus'
+```
+
+
+4. Add port forwarding for Prometheus & Grafana
```
Comment (XN137) [inline grafana/README.md:34]: nit: newlines between these numbers are inconsistent (here: 2, above: 0, below: 1)

File: grafana/README.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+```sh
+nohup kubectl -n nessie-dev port-forward svc/prometheus-kube-prometheus-prometheus 9090 &
+nohup kubectl -n nessie-dev port-forward svc/prometheus-grafana 3000:80 &
+```
```
Comment (XN137) [inline grafana/README.md:38]: nit: maybe also explain how to stop these + nessie + minikube again?

i also noticed that these port forwarding cmds seem much simpler than what you have to do when installing the nessie helmchart (i.e. the cmds it prints to find the right pods etc). is that something we could improve in the nessie helmchart as well?

File: grafana/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+1. Add Prometheus repo:
+```sh
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+```
+2. Install Prometheus
+```sh
+helm -n nessie-dev install prometheus prometheus-community/kube-prometheus-stack
+```
+3. Install Nessie (see also https://github.com/projectnessie/nessie/blob/main/helm/README.md)
```
Comment (XN137) [inline grafana/README.md:26]: nit: i noticed that the linked readme uses `nessie-ns` as namespace and `nessie-dev` in other places, maybe should be unified

File: grafana/README.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Prometheus & Grafana for Nessie
+
+## Dev installation for local experiments
+
+These steps are mainly for setting up Prometheus/Grafana/Nessie locally.
+
+* Install Minikube as described in https://minikube.sigs.k8s.io/docs/start/
+* Install Helm as described in https://helm.sh/docs/intro/install/
+* Start Minikube cluster: `minikube start`
+* Create K8s Namespace: `kubectl create namespace nessie-dev`
```
Comment (XN137) [inline grafana/README.md:10]: maybe all these steps for minikube and nessie could be avoided by just linking to https://github.com/projectnessie/nessie/blob/main/helm/README.md#dev-installation ?

**Review by XN137 — COMMENTED** [review]
i tried out the readme locally and it worked for me, see comments for some suggestions

File: grafana/README.md (hunk 4 above)
Comment (nastra) [reply grafana/README.md:8]: done

File: grafana/README.md (hunk 5 above)
Comment (nastra) [reply grafana/README.md:34]: fixed

File: site/docs/try/grafana.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+# Nessie Grafana Dashboard
+
+Nessie provides a [Grafana Dashboard](https://github.com/projectnessie/nessie/blob/main/grafana/nessie.json) which can be used
+to gain insight into different metrics. 
+Additional docs can be found [here](https://github.com/projectnessie/nessie/blob/main/grafana/README.md).
```
Comment (dimas-b) [inline site/docs/try/grafana.md:5]: nit: `here` -> `in the Nessie source code repository` (same link)?

File: grafana/README.md (hunk 7 above)
Comment (nastra) [reply grafana/README.md:25]: updated

File: grafana/README.md (hunk 6 above)
Comment (nastra) [reply grafana/README.md:36]: yep good point, updated

**Comment by snazy** [comment]
I'm happy with the dashboards, assuming those are the ones that you presented.

**Review by snazy — APPROVED** [review]

**Pull request merged**